CommonCompilerConfig.cmake

```diff
@@ -159,16 +183,24 @@ endif()
 # try compiling without them.
 #
 if(NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL "MSVC")
-  set(optimization_flags_cc "${optimization_flags_cc} -mtune=generic -O2")
+  set(optimization_flags_cc "${optimization_flags_cc} -O2")
   set(optimization_flags_cc "-pipe ${optimization_flags_cc}") #TODO use -Ofast instead of -O3
   set(optimization_flags_lt "-O2 ${optimization_flags_lt}")
 
+  CHECK_CXX_COMPILER_FLAG(-mtune=generic COMPILER_SUPPORTS_GENERIC_TUNING)
+  if(COMPILER_SUPPORTS_GENERIC_TUNING)
+    set(optimization_flags_cc "${optimization_flags_cc} -mtune=generic")
+  endif()
+
   if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU" AND NOT MINGW)
     set(optimization_flags_cc "${optimization_flags_cc} -fuse-ld=gold")
-    # NOTE -flto must go together in both cc and ld flags; also, it's presently incompatible
-    # with the -g option in at least some GNU compilers (saw in `man gcc` on Ubuntu)
-    set(optimization_flags_cc "${optimization_flags_cc} -fuse-linker-plugin -flto-report -flto") #TODO fix LTO for clang
-    set(optimization_flags_lt "${optimization_flags_lt} -flto") #TODO LTO for clang too
+
+    if(NOT "${PLATFORM_TYPE}" STREQUAL "arm")
```

Comment:
@breznak I'm unfamiliar with gold.  Are you saying that '-fuse-ld=gold' should go with the '-flto' and '-fuse-linker-plugin' options?  In other words, they should all be included or all be excluded?